### ISSUE TITLE
Documentation for Notifications API

### ISSUE THREAD

**Issue by BenWoodford** [ValveSoftware/openvr#395]
Some of it's fairly obvious (like the contents of NotificationBitmap_t, text, handle, etc) but the ones that aren't so obvious:

- ulUserValue
- type
- style

Been looking forward to having a notifications API as it's a pain to make them yourself.

**Comment by BenWoodford** [comment]
Addendum, both myself and others have been having problems using anything other than an empty bitmap on this. @

**Comment by BenWoodford** [comment]
Congrats to me for managing to hit comment on my phone before I was done.

@matzman666 care to chime in on this?

**Comment by matzman666** [comment]
Yeah, I also didn't get the NotificationBitmap_t to work. I tried to use GetOverlayImageData() to fill the NotificationBitmap_t with data from an existing overlay, but to no avail (My reasoning was that openVR uses GetOverlayImageData() to retrieve a default icon when you don't specify your own). The exact code I used to achieve this can be found [here](https://github.com/matzman666/OpenVR-AdvancedSettings/blob/master/src/tabcontrollers/UtilitiesTabController.cpp#L367-L390). 

An important point that is not clear at all is who takes ownership of the notification bitmap data. NotificationBitmap_t contains a void* pointer to the image data. When I pass the bitmap data to createNotification(), is the data copied and I can immediately free the buffer (that's what I tried to do and I didn't get a segmentation fault, so it seems plausible), or does OpenVR takes ownership of the buffer?

@BenWoodford 
You can find information about the type and style parameter [here](https://github.com/ValveSoftware/openvr/blob/7fa6470f2972970859f2395f1390f6d87d4b0fc3/headers/openvr.h#L2377-L2409). The ulUserValue seems to be used as an [unique id for system notifications](https://github.com/ValveSoftware/openvr/blob/7fa6470f2972970859f2395f1390f6d87d4b0fc3/headers/openvr.h#L2387), and it seems also to be [used in notification events](https://github.com/ValveSoftware/openvr/blob/7fa6470f2972970859f2395f1390f6d87d4b0fc3/headers/openvr.h#L683).

**Comment by BenWoodford** [comment]
That's helpful thanks, I use the C# header which doesn't have the comments, shall have to keep an eye on the .h file for documentation.

**Comment by BOLL7708** [comment]
I also use the C# wrapper but I never got notifications to work, I looked at it periodically over many months. Finally I used the working albeit somewhat outdated library [SteamVR_HUDCenter](https://github.com/artumino/SteamVR_HUDCenter). The author [mentions](https://github.com/artumino/SteamVR_HUDCenter/issues/2#issuecomment-285991529) that missing handles were manually added back to the wrapper to specifically expose notification functionality.

That would explain why notifications are tricky to get working in C# with the wrapper Valve provides, with the above mentioned library I have notifications with bitmaps working.
![vrcompositor_2017-04-17_21-15-30](https://cloud.githubusercontent.com/assets/12276837/26023630/128b8168-37c1-11e7-884d-f52e34ac5dbb.png)

**Comment by BenWoodford** [comment]
Yeah, I've got a similar addition. Curious as to how you got the right icon texture to work though?

**Comment by BOLL7708** [comment]
I actually checked the sample project in SteamVR_HUDCenter and used [this snippet](https://github.com/artumino/SteamVR_HUDCenter/blob/a8e306ba9c6fbe0e9834cd8d49365df42b06fa2e/VRTestApplication/TestForm.cs#L60-L76).

**Comment by BenWoodford** [comment]
Hmm. I did a similar thing [here](https://github.com/BenWoodford/SteamVR-Webkit/blob/master/SteamVR-WebKit/JsInterop/Notifications.cs) and it ended up showing me something irrelevant (the texture of the overlay I believe). Weird.
